server/controllers/user.ts

```diff
@@ -59,10 +59,14 @@ export function displayRegisterPage(req:Request, res: Response, next: NextFuncti
 }
 
 export function displayUserEditPage(req:Request, res: Response, next: NextFunction): void {
-    if(!req.user)
+
+    const id = req.params.id;
+
+    if(req.user?._id == id )
     {
-        res.render("index", {title: "EditUser", page: "edituser"});
+        return res.render("index", {title: "EditUser", page: "edituser"});
     }
+    return res.redirect("/logout");
```

Comment:
I don't think we would want to log out the user. Maybe just redirect to the account page?